trellis/virtualenv.go

```diff
@@ -0,0 +1,112 @@
+package trellis
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"trellis-cli/github"
+)
+
+const VirtualenvDir string = "virtualenv"
+const EnvName string = "VIRTUALENV"
+
+type Virtualenv struct {
+	Path    string
+	BinPath string
+}
+
+func NewVirtualenv(path string) *Virtualenv {
+	return &Virtualenv{
+		Path:    filepath.Join(path, VirtualenvDir),
+		BinPath: filepath.Join(path, VirtualenvDir, "bin"),
+	}
+}
+
+func (v *Virtualenv) Activate() {
+	os.Setenv(EnvName, v.Path)
+	os.Setenv("PATH", fmt.Sprintf("%s/bin:$PATH", v.Path))
+}
+
+func (v *Virtualenv) Active() bool {
+	return v.BinPath == os.Getenv(EnvName)
+}
+
+func (v *Virtualenv) Create() (err error) {
+	_, cmd := v.Installed()
+	cmd.Args = append(cmd.Args, v.Path)
+
+	if v.Initialized() {
+		return nil
+	}
+
+	err = cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	v.Activate()
+	return nil
+}
+
+func (v *Virtualenv) LocalPath() string {
+	configHome := os.Getenv("XDG_CONFIG_HOME")
+
+	if configHome == "" {
+		homeDir, err := os.UserHomeDir()
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		configHome = filepath.Join(homeDir, ".local", "share")
```

Comment:
Some developers perfer `~/Library/xxx` on macOS. What do you think?

See: https://stackoverflow.com/questions/3373948/equivalents-of-xdg-config-home-and-xdg-data-home-on-mac-os-x

Reply:
Interesting. There's a couple of replies/comments there that say don't use those for CLI based apps which I tend to agree with?